libhb/qsv_common.c

```diff
@@ -3558,7 +3558,7 @@ int hb_qsv_param_parse_dx_index(hb_job_t *job, const int dx_index)
                     return -1;
                 }
             }
-            sprintf(job->qsv.ctx->qsv_device, "%u", info->Number);
+            snprintf(job->qsv.ctx->qsv_device, 32, "%u", info->Number);
```

Comment:
This looks correct, though I wonder why device needs to be allocated in the first place? It might be just as good a time as any to make it fixed-size.